### ISSUE TITLE
Possible to disable just BarrelDistortion but not use the other features of this mod?

### ISSUE THREAD

**Issue by ezudo** [RyanJGray/aliasIsolation#16]
I was wondering if this was possible somehow, I'd like to just disable the barrel distortion effect but keep everything else vanilla. I'm planning on playing this with DSR/DLDSR applied, so there are almost no jaggies or shimmering edges to begin with, however since I also plan on using ReShade and some depth related effects, the barrel distortion really messes those up since the depth buffer does not align with what you actually see through the distorted lens effect.

Thank you

**Comment by whosnick** [comment]
> I'm planning on playing this with DSR/DLDSR applied, so there are almost no jaggies or shimmering edges to begin with

Really? I have tried this and even with like 8k to 4k it is still a shimmering mess.

**Comment by ezudo** [comment]
> 

They may not be completely gone, but from my experience **DLDSR** does a better job getting rid of those issues compared to regular DSR since it uses some AI image reconstruction stuff on top of downsampling from what I understand. In any case, I'm happy enough with the result :)

**Comment by RyanJGray** [comment]
> I was wondering if this was possible somehow, I'd like to just disable the barrel distortion effect but keep everything else vanilla. I'm planning on playing this with DSR/DLDSR applied, so there are almost no jaggies or shimmering edges to begin with, however since I also plan on using ReShade and some depth related effects, the barrel distortion really messes those up since the depth buffer does not align with what you actually see through the distorted lens effect.
> 
> Thank you

Hey there!

It should be possible to use this just to disable the barrel distortion effect.
I'll see about adding an option for that. I imagine it should be relatively straightforward, as one part of the mod, replaces the game's main vertex shader, with one that removes the effect.

I'll have a look into this for you, hopefully will have a solution in the next day or two.

**Comment by ezudo** [comment]
> > I was wondering if this was possible somehow, I'd like to just disable the barrel distortion effect but keep everything else vanilla. I'm planning on playing this with DSR/DLDSR applied, so there are almost no jaggies or shimmering edges to begin with, however since I also plan on using ReShade and some depth related effects, the barrel distortion really messes those up since the depth buffer does not align with what you actually see through the distorted lens effect.
> > Thank you
> 
> Hey there!
> 
> It should be possible to use this just to disable the barrel distortion effect. I'll see about adding an option for that. I imagine it should be relatively straightforward, as one part of the mod, replaces the game's main vertex shader, with one that removes the effect.
> 
> I'll have a look into this for you, hopefully will have a solution in the next day or two.

Awesome! Thank you, looking forward to it.

**Comment by ezudo** [comment]
Did you find any time to look into this yet? I'm waiting for this before I start my playthrough, however please don't feel pressured or anything, it's not an issue really, plenty of other game backlog available to play :)

**Comment by RyanJGray** [comment]
> Did you find any time to look into this yet? I'm waiting for this before I start my playthrough, however please don't feel pressured or anything, it's not an issue really, plenty of other game backlog available to play :)

Not at the moment, but I'll have some time this week.

I might honestly split it out into its own tool, as you only really need the barrel distortion shader instead of the entire TAA mod.

Bear with me! (:

**Comment by ezudo** [comment]
Any news on this?

**Comment by RyanJGray** [comment]
> Any news on this?

I'm working on adding a new menu into the mod in-game at the moment, that will allow you to basically disable every other part of the mod (in terms of shaders) except for the barrel distortion fix shader.

It'll be in v1.2.0, I want to get the mod moved over to the main aliasIsolation repository before I add more features.

I'll keep you posted!

**Comment by Klemci** [comment]
> > I'm planning on playing this with DSR/DLDSR applied, so there are almost no jaggies or shimmering edges to begin with
> 
> Really? I have tried this and even with like 8k to 4k it is still a shimmering mess.

Yes, only Alias renders well.

https://forums.guru3d.com/threads/nvidia-anti-aliasing-guide-updated.357956/page-262#post-6203766

**Comment by brad86m** [comment]
> > Any news on this?
> 
> I'm working on adding a new menu into the mod in-game at the moment, that will allow you to basically disable every other part of the mod (in terms of shaders) except for the barrel distortion fix shader.
> 
> It'll be in v1.2.0, I want to get the mod moved over to the main aliasIsolation repository before I add more features.
> 
> I'll keep you posted!

Still in the works ? Would love to try a newer version of your Alias mod. Currently, the older menu toggle not work for me, nor does adding entries to a settings files. Really want the CA and barrel distortion disabled like the original mod did.

**Comment by RyanJGray** [comment]
Apologies for the long wait on this request. There's a lot going on at work these days and tinkering in the background on Alias.

Barrel distortion is, by default, always removed in Alias via a shader, I don't have a toggle for it at the moment. I think it may interfere with the mod if the barrel distortion effect remains

CA can be turned off completely in the latest release of Alias. What issue specifically are you having with the menu? Could you maybe open a new issue for that?

The new menu in this fork is accessed via `DELETE` on the keyboard. It should work, but for Steam Deck users without a keyboard, it will be inaccessible for the time being until I add gamepad controls
